### DISCUSSION TITLE
Symbol # in password leads to "Connection refused"

### DISCUSSION THREAD

**Discussion by EBCEEB** [rabbitmq/rabbitmq-perf-test#613] [General]
### Describe the bug

I have `#` in password and can't connect to a RabbitMQ

### Reproduction steps

1. Create user having `#` in password
2. Run perf test with uri like `amqp://user:#@host:5672/`

### Expected behavior

Sussecful connect and test

**Comment by michaelklishin** [comment]
PerfTest is a CLI tool, so any characters that may be treated specially by shells must be treated with extra care in the arguments.

Neither RabbitMQ itself nor its client libraries treat `#` as a special character. Your shell does.
Without a specific example of how **exactly** you run it and in what shell on what OS, we can only guess what may be going on.

**Comment by michaelklishin** [comment]
``` shell
# the password value MUST be enquoted or else the `#` can be interpreted as a comment by the shell
rabbitmqctl add_user "perftest-hash-pwd" "perftest-@-hash-#-pwd"
rabbitmqctl set_permissions "perftest-hash-pwd" ".*" ".*" ".*"

java -jar ./perf-test-2.19.0.jar --uri 'amqp://perftest-hash-pwd:perftest-@-hash-#-pwd@localhost'
```

works on my machine.

RabbitMQ's own documentation about seeding users [mentions this and other characters](https://rabbitmq.com/access-control.html#passwords-and-shell-escaping) in a dedicated section.
